### PR TITLE
Add s3 support for --file flag

### PR DESCRIPTION
**Pull request by flatnine**
Adjusts `treehugger` so the --file flag can be used within Containers to retrieve files from S3 as UserData is not available.

### CODE REVIEW

File: tests/test_yaml.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# -*- coding:utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from treehugger.yaml import _split_s3_filename
+
+
+class TestYaml:
```
Comment (adamchainz) [inline tests/test_yaml.py]: test class isn't required

File: tests/test_yaml.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+    def test_split_s3_filename(self):
+        test_s3_filename = 's3://bucket/key?versionId=7'
+        assert _split_s3_filename(test_s3_filename) == ('bucket', 'key', '7')
```
Comment (adamchainz) [inline tests/test_yaml.py]: `read_file_from_s3` could be tested if you set up with `botocore.stub.Stubber` like other tests

Comment (flatnine) [reply tests/test_yaml.py]: Don't worry already on it.  Couldn't figure out how to request your input on this as you aren't a collaborator.

BTW I merged your requirements changes but how do we publish to PyPi?  I see the last release was in Jan.   I plan to start reading up on the process later.

Comment (adamchainz) [reply tests/test_yaml.py]: You can always @ anyone on github, like @adamchainz . Or just email me!

Oh woops, I'm the only maintainer on PyPI: https://pypi.org/project/treehugger/ . Email me when you've set up a PyPI account.

**Comment by flatnine** [comment]
Check import for 2.7 . can't use 

from .yaml import safe_mode 

https://github.com/timeoutdigital/treehugger/pull/117/files#diff-4d132ca184e719c5d0eb15006e08ada3R9

in python 2.7? 
Is this because of yaml.py already imports yaml and overrides safe_load?  

Read about imports

**Comment by jbasko** [comment]
@flatnine I think it's because of a circular import:

    File "/home/travis/build/timeoutdigital/treehugger/.tox/py27/local/lib/python2.7/site-packages/treehugger/ec2.py", line 10, in <module>
    from .yaml import safe_load
    File "/home/travis/build/timeoutdigital/treehugger/.tox/py27/local/lib/python2.7/site-packages/treehugger/yaml.py", line 10, in <module>
    from .s3 import read_file_from_s3
    File "/home/travis/build/timeoutdigital/treehugger/.tox/py27/local/lib/python2.7/site-packages/treehugger/s3.py", line 9, in <module>
    from . import yaml

**Comment by adamchainz** [comment]
🔄

**Comment by flatnine** [comment]
@adamchainz Reusing the include functionality was how I originally got treehugger running but it just felt ugly, though that may have just been my implementation.

**Comment by adamchainz** [comment]
I think it is more elegant than adding a niche option, and it reuses the code in `include_remote_yaml_data_or_die` for loading data from s3.

**Pull request merged**